version: '3.8'

services:
  mysql:
    image: mysql:8
    container_name: mysql-kulinaria
    environment:
      MYSQL_ROOT_PASSWORD: rootpassword
      MYSQL_DATABASE: kulinaria
      MYSQL_USER: appuser
      MYSQL_PASSWORD: tajnehaslo
    ports:
      - "3306:3306"
    volumes:
      - mysql_data:/var/lib/mysql
      - ./db-init:/docker-entrypoint-initdb.d
    networks:
      - kulinaria-network
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      timeout: 20s
      retries: 10

  app:
    build: .
    container_name: zti-app
    ports:
      - "8080:8080"
    depends_on:
      mysql:
        condition: service_healthy
    networks:
      - kulinaria-network
    environment:
      - DB_HOST=mysql
      - DB_PORT=3306
      - DB_NAME=kulinaria
      - DB_USER=appuser
      - DB_PASSWORD=tajnehaslo

volumes:
  mysql_data:

networks:
  kulinaria-network:
    driver: bridge 